auslib/db.py

```diff
@@ -1880,14 +1890,15 @@ def hasPermission(self, username, thing, action, product=None, transaction=None)
         except ValueError:
             return False
 
-        # If a user has a permission that doesn't explicitly limit the type of
-        # actions they can perform, they are allowed to do any type of action.
-        if options.get("actions") and action not in options["actions"]:
-            return False
-        # Similarly, permissions without products specified grant that
-        # that permission without any limitation on the product.
-        if options.get("products") and product not in options["products"]:
-            return False
+        if options:
+            # If a user has a permission that doesn't explicitly limit the type of
+            # actions they can perform, they are allowed to do any type of action.
+            if options.get("actions") and action not in options["actions"]:
+                return False
+            # Similarly, permissions without products specified grant that
+            # that permission without any limitation on the product.
+            if options.get("products") and product not in options["products"]:
+                return False
 
         return True
```

Comment:
What do you think of this @nthomas-mozilla ? I'm happy to include it in my branch if we're all agreed on it.